src/crypto/shelley/metadataUtils.js

```diff
@@ -8,23 +8,23 @@ import {
   MetadataJsonSchema,
   GeneralTransactionMetadata,
   BigNum,
-  TransactionMetadata,
+  AuxiliaryData,
 } from '@emurgo/react-native-haskell-shelley'
 
 export type JSONMetadata = {|
   label: string,
   data: {},
 |}
 
-export async function createMetadata(metadata: Array<JSONMetadata>): TransactionMetadata {
-  const transactionMetadata = await GeneralTransactionMetadata.new()
+export async function createAuxiliaryData(auxiliary: Array<JSONMetadata>): AuxiliaryData {
+  const metadata = await GeneralTransactionMetadata.new()
 
-  for (const meta of metadata) {
+  for (const meta of auxiliary) {
     const metadatum = await encode_json_str_to_metadatum(JSON.stringify(meta.data), MetadataJsonSchema.BasicConversions)
-    await transactionMetadata.insert(await BigNum.from_str(meta.label), metadatum)
+    await metadata.insert(await BigNum.from_str(meta.label), metadatum)
   }
 
-  return await TransactionMetadata.new(transactionMetadata)
+  return await AuxiliaryData.new(metadata)
```

Comment:
Hey @yushih, you can't rely on flow for the bindings because is not 100% covered, and it is been ignored for now. You are correct, the implementations are different, nice catch. You can check it on https://github.com/Emurgo/react-native-haskell-shelley/blob/develop-alonzo/rust/src/android/auxiliary_data.rs#L56 it is not on main yet.